root/static/scripts/common/components/ReleaseEvents.js

```diff
@@ -18,6 +18,8 @@ import isDateEmpty from '../utility/isDateEmpty';
 import EntityLink from './EntityLink';
 
 const COLLAPSE_THRESHOLD = 4;
+const BEFORE_COLLAPSE = 2;
+const AFTER_COLLAPSE = 1;
```

Comment:
Thanks for documenting hard-coded values but I can hardly understand the exact meaning of these constants.

I would suggest to:
1. Replace `THRESHOLD` (which is a bit ambiguous: collapse at it or after it?)
   with `TRIGGER` which is less ambiguous to me but must be set to 5 instead of 4.
2. Replace `BEFORE_COLLAPSE`
   with `EXPAND_ITEM_HEIGHT` which is the height (in number of normal items, rounded up) taken by the expand item.
3. Drop `AFTER_COLLAPSE` which might not be needed after the two above replacements.

```suggestion
const COLLAPSE_TRIGGER = 5;
const EXPAND_ITEM_HEIGHT = 2;
```

The rest of the code would need to be adjusted accordingly.

Reply:
I think I got it: `BEFORE_COLLAPSE` and `AFTER_COLLAPSE` are for the number of items that are displayed resp. above and below the (expand) `show-all` item. Thus the above suggestions 2. and 3. make lesser sense, even though I would still rename these two added constants to `ABOVE_EXPAND_ITEM` and `BELOW_EXPAND_ITEM` resp. (_expand_ or _show all_).

I just noticed that artists are not collapsed the same way as release events: No artist is displayed below the `show-all` item, and the `show-all` item height is not taken into account. Should we uniformize to release events’ collapsed display (2+1) rather than artists’ collapsed display (4+)?

Reply:
I would say no because the split has a clear purpose on release events ("this are the first and last releases, by date") but it wouldn't have on artists (I'm not sure even how those are sorted, just alphabetically possibly?)

Reply:
I didn't initially intend for the number of items before & after to be configurable. `COLLAPSE_THRESHOLD` was just the max number of items before it collapsed things...so:

 * If COLLAPSE_THRESHOLD=4 it'll show 4, but if there's 5 it'll show 2 + (2 hidden) + 1
 * If COLLAPSE_THRESHOLD=5 it'll show 5, but if there's 6 it'll show 3 + (2 hidden) + 1
 * etc.

So if we decided we wanted to allow showing more initially, we could just update `COLLAPSE_THRESHOLD` and that's it. That still seems ideal to me, though a constant for the "2 hidden" might still be useful (I suggested `COLLAPSE_MIN`) above.

Reply:
It’s more about documenting code through constants' name than allowing to configure.

* _threshold_ is ambiguous because it does not specify it actually is an upper bound (that is, nothing happens up to 4) and not a lower bound (which would be the suggested _trigger_ replacement set to 5).

* _collapse_before_ and _collapse_after_ are misleading because it is about `show-all` button to expand the list, not to collapse it (hence the suggested replacements).

Reply:
Agree with @yvanzo here. I feel the constants make the code much more clear. I went with this in the latest version of the commit:

```
const TO_TRIGGER_COLLAPSE = 4;
const TO_SHOW_BEFORE = 2;
const TO_SHOW_AFTER = 1;
```
I think those are the clearer names for the constants here.

Reply:
> `const TO_TRIGGER_COLLAPSE = 4;`

Should be `5` instead of `4` though, right?